src/main/java/com/github/games647/craftapi/cache/SafeCacheBuilder.java

```diff
@@ -149,18 +149,21 @@ private SafeCacheBuilder() {
      * @throws IllegalStateException    if a maximum size was already set
      */
     public SafeCacheBuilder<K, V> maximumSize(int size) {
+        boolean isLong;
         try {
             MAXIMUM_SIZE_METHOD = builder.getClass().getDeclaredMethod("maximumSize", Long.TYPE);
+            isLong = true;
         } catch (NoSuchMethodException noSuchMethodEx) {
             try {
                 MAXIMUM_SIZE_METHOD = builder.getClass().getDeclaredMethod("maximumSize", Integer.TYPE);
+                isLong = false;
             } catch (NoSuchMethodException e) {
                 throw new IllegalStateException("Unable to find CacheBuilder.maximumSize(Int|Long)", e);
             }
         }
 
         try {
-            MAXIMUM_SIZE_METHOD.invoke(builder, size);
+            MAXIMUM_SIZE_METHOD.invoke(builder, isLong?(Long)size:(Integer)size);
```

Comment:
Please reformat this code and Integer cannot be casted to Longs (You can only cast them if you are using the primitive types - lower case). Furthermore the type has to be known at compile time, so we need to separate the conditions into different if stateents.
```suggestion
            if (isLong) {
                MAXIMUM_SIZE_METHOD.invoke(builder, (long) size);
            } else {
                MAXIMUM_SIZE_METHOD.invoke(builder, size);
            }
```

Reply:
fixed.
I've just compiled and tested it and found this mistake.
It hasn't to be separated since the type is `Object`, not `Long` or `Integer`

Reply:
Yes, but you will get runtime errors. Could please try to run the unit tests.

`mvn test`

Reply:
fixed